src/backend/web/routes/auth.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express');
+const passport = require('passport');
+
+const { logger } = require('../../utils/logger');
+
+const router = express.Router();
+const telescopeHomeUrl = '/';
+
+/**
+ * /auth/login/callback is where the external SAML SSO provider will redirect
+ * users upon successful login.  We redirect the user back to wherever
+ * they were trying to go, or to the home page
+ */
+router.post('/login/callback', passport.authenticate('saml'), (req, res) => {
+  let returnTo;
+  if (req.session) {
+    returnTo = req.session.returnTo;
+    delete req.session.returnTo;
+  }
+
+  res.redirect(returnTo || telescopeHomeUrl);
```

Comment:
interesting - i was just saying that seneca works needs this ability.

Reply:
Yeah, it's really critical, and you have to hack it in place yourself.